feat: add support for more third party backend formats under Kobold settings

malfoyslastname opened the pull request:
Makes the "Kobold" settings able to support more third-party backend formats than just Kobold. This commits adds support for backends using the OpenAI format. In the future more will be added, e.g. ooba.

This adds a new "Third party backend format" under the Kobold settings tab. If "OpenAI" is chosen then requests will be made using the OpenAI adapter, but with the provided backend URL as a baseUrl.

The name for the "Kobold" settings category is no longer fully accurate but we've deferred naming issues for now.

![1681574463](https://user-images.githubusercontent.com/128472336/232235790-2f876a6a-7566-4873-8db3-810f5fa37a1a.png)

catboxanon commented:
imo, this really should be also reading from the env vars like the Python lib does (`OPENAI_API_BASE`). The setting in the UI could act as an override.

https://github.com/openai/openai-python/blob/d6fa3bfaae69d639b0dd2e9251b375d7070bbef1/openai/__init__.py#L48

The API key should be handled the same way (`OPENAI_API_KEY`).

malfoyslastname commented:
@sceuick Can we prioritize this as its usefulness peaks now and might sharply decline soon

Commit:
feat: Support multiple third-party backend formats

Makes the "Kobold" settings able to support more third-party backend formats than just Kobold.

This commits adds support for backends using the OpenAI format. In the
future more will be added, e.g. ooba.

malfoyslastname commented:
@sceuick force pushed with different way to store this setting, as discussed

Commit:
fix: remove console log

Commit:
feat: add claude-formatted API support for third party backends

malfoyslastname commented:
[58767e3](https://github.com/luminai-companion/agn-ai/pull/172/commits/58767e3bced3db3d6aefc9cce5b0db3376af75c7) adds support for claude-compatible APIs

sceuick committed:
- Strip keys
- Rename user props
- Simplify base url logic

sceuick committed:
Enable pipeline on PRs

sceuick committed:
Default 3rd party format to kobold
